ops/modules/ecr/main.tf

```diff
@@ -12,12 +12,52 @@ resource "aws_ecr_lifecycle_policy" "remove_old_images" {
     "rules": [
         {
             "rulePriority": 1,
-            "description": "Expire images older than 20 days",
+            "description": "Expire main images that are not the last 50",
             "selection": {
-                "tagStatus": "any",
+                "tagStatus": "tagged",
+                "tagPrefixList": ["main-"],
+                "countType": "imageCountMoreThan",
+                "countNumber": 50
+            },
+            "action": {
+                "type": "expire"
+            }
+        },
+        {
+            "rulePriority": 2,
+            "description": "Expire staging images that are not the last 20",
+            "selection": {
+                "tagStatus": "tagged",
+                "tagPrefixList": ["staging-"],
+                "countType": "imageCountMoreThan",
+                "countNumber": 20
+            },
+            "action": {
+                "type": "expire"
+            }
+        },
+        {
+            "rulePriority": 3,
+            "description": "Expire prod images that are not the last 5",
+            "selection": {
+                "tagStatus": "tagged",
+                "tagPrefixList": ["prod-"],
```

Comment:
Can we differentiate between `testnet-prod` and `prod` ?

Reply:
That'd be together with `testnet-` - I don't think we need to differentiate them. Wdyt?

Reply:
ic, that works, and it is easy to change if we need to.